src/plugins/mf_priority.cpp

```diff
@@ -1116,6 +1144,42 @@ extern "C" int flux_plugin_init (flux_plugin_t *p)
     priority_weights["queue"] = DEFAULT_QUEUE_WEIGHT;
     priority_weights["age"] = DEFAULT_AGE_WEIGHT;
 
+    // initialize the plugin with total node and core counts
+    flux_t *h;
+    flux_future_t *f;
+    char *core;
+
+    h = flux_jobtap_get_flux (p);
+    if (!(f = flux_kvs_lookup (h,
+                               NULL,
+                               FLUX_KVS_WATCH | FLUX_KVS_WAITCREATE,
+                               "resource.R"))) {
+        flux_log_error (h, "flux_kvs_lookup");
+        return -1;
+    }
+
+    if (flux_kvs_lookup_get_unpack (f,
+                                    "{s{s[{s{s:s}}]}}",
+                                    "execution",
+                                      "R_lite",
+                                        "children",
+                                          "core", &core) < 0) {
```

Comment:
R may have multiple entries in the R_lite array. To handle possible heterogeneity,  you could iterate each entry and use the maximum number of cores found. 

Probably ok if this is just a first cut, though. If so, I'd put a comment stating that "equal number of cores on all nodes in R is assumed, so we only look at the first entry" or simimlar.